### PR TITLE
TST: Split/parameterize test_split_partition.py

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them

### CODE REVIEW

File: pandas/tests/strings/test_split_partition.py [hunk 1]
```diff
@@ -92,81 +103,42 @@ def test_split_object_mixed():
...
-    result = getattr(s.str, method)(" ", n=None)
-    tm.assert_series_equal(result, expected)
-
-    result = getattr(s.str, method)(" ", n=0)
+    result = getattr(s.str, method)(" ", n=n)
     tm.assert_series_equal(result, expected)
 
 
 def test_rsplit(any_string_dtype):
-    values = Series(["a_b_c", "c_d_e", np.nan, "f_g_h"], dtype=any_string_dtype)
```
Comment (jreback) [inline pandas/tests/strings/test_split_partition.py:113]: don't you need these cases?

File: pandas/tests/strings/test_split_partition.py [hunk 2]
```diff
@@ -92,81 +103,42 @@ def test_split_object_mixed():
...
+
+def test_rsplit_max_number(any_string_dtype):
     # setting max number of splits, make sure it's from reverse
     values = Series(["a_b_c", "c_d_e", np.nan, "f_g_h"], dtype=any_string_dtype)
     result = values.str.rsplit("_", n=1)
     exp = Series([["a_b", "c"], ["c_d", "e"], np.nan, ["f_g", "h"]])
     tm.assert_series_equal(result, exp)
 
 
-def test_rsplit_object_mixed():
```
Comment (jreback) [inline pandas/tests/strings/test_split_partition.py:139]: where did these go>?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/strings/test_split_partition.py (hunk 2 above)
Comment (mroeschke) [reply pandas/tests/strings/test_split_partition.py:139]: Parameterized now in `test_split_object_mixed`

File: pandas/tests/strings/test_split_partition.py (hunk 1 above)
Comment (mroeschke) [reply pandas/tests/strings/test_split_partition.py:113]: Parameterized now in `test_split` and  `test_split_more_than_one_char`,

**Review by jreback — APPROVED** [review]

File: pandas/tests/strings/test_split_partition.py [hunk 3]
```diff
@@ -419,191 +409,210 @@ def test_split_with_name(any_string_dtype):
...
     # https://github.com/pandas-dev/pandas/issues/23558
-
-    s = Series(["a_b_c", "c_d_e", np.nan, "f_g_h", None], dtype=any_string_dtype)
-
-    result = s.str.partition("_", expand=False)
-    expected = Series(
-        [("a", "_", "b_c"), ("c", "_", "d_e"), np.nan, ("f", "_", "g_h"), None]
-    )
+    # more than one char
+    s = Series(["a__b__c", "c__d__e", np.nan, "f__g__h", None])
```
Comment (simonjayhawkins) [inline pandas/tests/strings/test_split_partition.py:440]: `any_string_dtype` parameterization?

**Pull request merged**